packages/connect-webextension/src/proxy/index.ts

```diff
@@ -0,0 +1,127 @@
+import EventEmitter from 'events';
+
+// NOTE: @trezor/connect part is intentionally not imported from the index due to NormalReplacementPlugin
+// in packages/suite-build/configs/web.webpack.config.ts
+import {
+    ERRORS,
+    IFRAME,
+    createErrorMessage,
+    ConnectSettings,
+    Manifest,
+    CallMethod,
+} from '@trezor/connect/lib/exports';
+import { factory } from '@trezor/connect/lib/factory';
+
+import { parseConnectSettings } from '../connectSettings';
+import { WindowServiceWorkerChannel } from '../channels/window-serviceworker';
+
+const eventEmitter = new EventEmitter();
+let _settings = parseConnectSettings();
+
+let _channel: any;
+
+const manifest = (data: Manifest) => {
+    _settings = parseConnectSettings({
+        ..._settings,
+        manifest: data,
+    });
+};
+
+const dispose = () => {
+    eventEmitter.removeAllListeners();
+    _settings = parseConnectSettings();
+
+    if (_channel) {
+        _channel.clear();
+    }
+    return Promise.resolve(undefined);
+};
+
+const cancel = () => {
+    if (_channel) {
+        // TODO: should we implement other cancelation here?
+        _channel.clear();
+    }
+};
+
+const init = async (settings: Partial<ConnectSettings> = {}): Promise<void> => {
+    _settings = parseConnectSettings({ ..._settings, ...settings });
+
+    if (!_channel) {
+        _channel = new WindowServiceWorkerChannel({
+            name: 'trezor-connect-proxy',
+            channel: {
+                here: '@trezor/connect-foreground-proxy',
+                peer: '@trezor/connect-service-worker-proxy',
+            },
+        });
+    }
+
+    if (!_settings.manifest) {
+        throw ERRORS.TypedError('Init_ManifestMissing');
+    }
+
+    if (!_settings.transports?.length) {
+        _settings.transports = ['BridgeTransport', 'WebUsbTransport'];
+    }
+
+    return _channel.init();
+};
+
+const call: CallMethod = async (params: any) => {
+    try {
+        const response = await _channel.postMessage({
+            type: IFRAME.CALL,
+            payload: params,
+        });
+        if (response) {
+            return response;
+        }
+        return createErrorMessage(ERRORS.TypedError('Method_NoResponse'));
+    } catch (error) {
+        _channel.clear();
+
+        return createErrorMessage(error);
+    }
+};
+
+const uiResponse = () => {
+    // TODO: ???
+};
+
+const renderWebUSBButton = () => {
+    // TODO: ???
+};
+
+const requestLogin = () => {
+    // TODO: not supported yet.
+    throw ERRORS.TypedError('Method_InvalidPackage');
+};
```

Comment:
@mroz22 I am not sure if those should be somehow defined in this proxy.

Reply:
imho yes, proxy should have the same interface as every connect package. simply throwing error on them is enough imho 🤔